File: backend/src/main/java/com/cdp/ecosaas/procurement/supplier/application/command/CreateSupplierCommand.java
package com.cdp.ecosaas.procurement.supplier.application.command;

import com.cdp.ecosaas.procurement.supplier.domain.model.SupplierCategory;

/**
 * 创建供应商命令（Req 6.1-6.3）。
 * <p>
 * 携带企业名称、分类与主要联系人（姓名/手机号/邮箱）；{@code sendInvitation} 区分
 * 「保存并发送邀请」（true→发邀请邮件、状态置「待进入」）与「仅保存」（false→状态「创建成功」）。
 */
public record CreateSupplierCommand(
        String name,
        SupplierCategory category,
        String contactName,
        String contactPhone,
        String contactEmail,
        boolean sendInvitation
) {
}

File: frontend/src/modules/supplier/types/command/create-supplier.command.ts
import type { SupplierCategory } from '../vo/supplier-info.vo';

/**
 * 创建供应商命令（采购端，Req 6.1-6.3）。
 * <p>
 * sendInvitation=true 为「保存并发送邀请」（→待进入）；false 为「仅保存」（→创建成功）。
 */
export interface CreateSupplierCommand {
  name: string;
  category: SupplierCategory;
  contactName: string;
  contactPhone: string;
  contactEmail: string;
  sendInvitation: boolean;
}
